scanner/scanner.go

```diff
@@ -0,0 +1,168 @@
+package scanner
+
+import (
+	"bufio"
+	"bytes"
+	"io"
+)
+
+// constants
+const (
+	EOF = iota
```

Comment:
nit: You could use a string Consts for these instead of ints.

You also might provide a custome type here for clarity.

type ControlChar int

or similar

Or maybe

type ControlChar rune